add support for created and expires values

blacktemplar opened the pull request:
Resolves #109.

Adds support for created and expires values in parser and signer.

zetxx commented:
+1 is anyone on this ?

jasonbking commented:
I'll try to get this reviewed sometime the next several days..

jasonbking commented on lib/signer.js:
```diff
@@ -41,9 +41,13 @@ function FormatAuthz(prefix, params) {
       continue;
-    assert.string(value, 'params.' + param);
+    if (typeof value === 'number') {
+      authz += prefix + sprintf('%s=%s', param, value);
```
Shouldn't this be `%s=%d` or `%s=%u`?

blacktemplar replied:
It works both (`%s` and `%d`), since `%s` automatically converts the number to a string (see https://nodejs.org/api/util.html#util_util_format_format_args). `%u` is not documented, I will update this to `%s` to better signal that value is a number.

blacktemplar commented:
I fixed all `make check` errors and added some new tests, testing the parsing behavior.

`make test` runs through except for one test (not ok 10 test/examples.test.js) which I think is unrelated to my pull request and also fails on my machine on the master branch.

But the CI still fails, so I am not sure whats the fault.

PS: would be good to document somewhere that one needs to install the npm package json globally to use `make check`, at least I didn't find it anywhere and needed some googling.
